cmd/storage-rest-server.go

```diff
@@ -1083,7 +1083,7 @@ func registerStorageRESTHandlers(router *mux.Router, endpointServerSets Endpoint
 			subrouter.Methods(http.MethodPost).Path(storageRESTVersionPrefix + storageRESTMethodVerifyFile).HandlerFunc(httpTraceHdrs(server.VerifyFileHandler)).
 				Queries(restQueries(storageRESTVolume, storageRESTFilePath)...)
 			subrouter.Methods(http.MethodPost).Path(storageRESTVersionPrefix + storageRESTMethodWalkDir).HandlerFunc(httpTraceHdrs(server.WalkDirHandler)).
-				Queries(restQueries(storageRESTVolume, storageRESTDirPath, storageRESTRecursive, storageRESTPrefixFilter)...)
+				Queries(restQueries(storageRESTVolume, storageRESTDirPath, storageRESTRecursive)...)
```

Comment:
This means that storageRESTPrefixFilter value won't be passed to the handler even if specified, you cannot make it optional, that's how mux works.

Reply:
@vadmeste you can make it optional storageRESTPrefixFilter will always be passed what you need to make sure is not look for `mux.Vars` instead use `r.URL.Query()`